### PR TITLE
Replace sprintf with snprintf in java.base

### PR DESCRIPTION
**Pull request by pshipton**
In files:
check_version.c
jvm_dtrace.c
libjvm_db.c
ResolverConfigurationImpl.c

### CODE REVIEW

**Comment by pshipton** [comment]
There are more but they will need to be fixed in later versions first before backporting.

File: src/java.base/solaris/native/libjvm_db/libjvm_db.c [hunk 1]
```diff
@@ -1327,7 +1332,7 @@ int Jget_vframe(jvm_agent_t* J, int vframe_no,
   int32_t   err;
 
   if (vframe_no >= N->vf_cnt) {
-     (void) sprintf(name, "Wrong inlinedMethod%1d()", vframe_no);
+     (void) snprintf(name, size, "Wrong inlinedMethod%1d()", vframe_no);
```
Comment (keithc-ca) [inline src/java.base/solaris/native/libjvm_db/libjvm_db.c:1335]: While you're here, please correct the size argument on line 1340:
```c
  err = name_for_methodPtr(J, vf->method, name + 1, size - 1);
```
(see lines 1097 and 1100 for comparison).

**Review by keithc-ca — CHANGES_REQUESTED** [review]

File: src/java.base/solaris/native/libjvm_db/libjvm_db.c (hunk 1 above)
Comment (pshipton) [reply src/java.base/solaris/native/libjvm_db/libjvm_db.c:1335]: Updated.

**Comment by keithc-ca** [comment]
Jenkins test sanity.functional xlinux,win jdk11

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**